pkg/teleporter/relayer.go

```diff
@@ -0,0 +1,377 @@
+		toFund := big.NewInt(0).Sub(teleporterRelayerRequiredBalance, teleporterRelayerBalance)
+		err := evm.FundAddress(
+			client,
+			prefundedPrivateKey,
+			teleporterRelayerAddress,
+			toFund,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type relayerRunFile struct {
+	Pid int `json:"pid"`
+}
+
+func DeployRelayer(
+	binDir string,
+	configPath string,
+	logFilePath string,
+	runFilePath string,
+	storageDir string,
+) error {
+	if err := RelayerCleanup(runFilePath, storageDir); err != nil {
+		return err
+	}
+	downloader := application.NewDownloader()
+	version, err := downloader.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(constants.AvaLabsOrg, constants.AWMRelayerRepoName))
+	if err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("using latest awm-relayer version (%s)", version)
+	versionBinDir := filepath.Join(binDir, version)
+	binPath, err := installRelayer(versionBinDir, version)
+	if err != nil {
+		return err
+	}
+	pid, err := executeRelayer(binPath, configPath, logFilePath)
+	if err != nil {
+		return err
+	}
+	return saveRelayerRunFile(runFilePath, pid)
+}
+
+func RelayerCleanup(runFilePath string, storageDir string) error {
+	if err := os.RemoveAll(storageDir); err != nil {
+		return err
+	}
+	if !utils.FileExists(runFilePath) {
+		return nil
+	}
+	bs, err := os.ReadFile(runFilePath)
+	if err != nil {
+		return err
+	}
+	rf := relayerRunFile{}
+	if err := json.Unmarshal(bs, &rf); err != nil {
+		return err
+	}
+	proc, err := os.FindProcess(rf.Pid)
+	if err != nil {
+		// so much expected after a reboot
+		if err := os.Remove(runFilePath); err != nil {
+			return fmt.Errorf("failed removing relayer run file %s: %w", runFilePath, err)
+		}
+		return nil
+	}
+	if err := proc.Signal(syscall.Signal(0)); err != nil {
+		// so much expected after a reboot
+		if err := os.Remove(runFilePath); err != nil {
+			return fmt.Errorf("failed removing relayer run file %s: %w", runFilePath, err)
+		}
+		return nil
+	}
+	if err := proc.Signal(os.Interrupt); err != nil {
+		return fmt.Errorf("failed killing relayer process with pid %d: %w", rf.Pid, err)
+	}
+	if err := os.Remove(runFilePath); err != nil {
+		return fmt.Errorf("failed removing relayer run file %s: %w", runFilePath, err)
+	}
+	return nil
+}
+
+func saveRelayerRunFile(runFilePath string, pid int) error {
+	rf := relayerRunFile{
+		Pid: pid,
+	}
+	bs, err := json.Marshal(&rf)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(runFilePath, bs, constants.WriteReadReadPerms); err != nil {
+		return fmt.Errorf("could not write awm relater run file to %s: %w", runFilePath, err)
+	}
+	return nil
+}
+
+func installRelayer(binDir, version string) (string, error) {
+	binPath := filepath.Join(binDir, constants.AWMRelayerBin)
+	if utils.IsExecutable(binPath) {
+		return binPath, nil
+	}
+	ux.Logger.PrintToUser("Installing AWM-Relayer %s", version)
+	url, err := getRelayerURL(version)
```

Comment:
should `getRelayerURL` return file ext so it can be passed instead of hardcoding `"tar.gz"`

Reply:
not needed, always tar.gz